047-TrafficControlWithDapr/Student/Challenge-00.md

```diff
@@ -0,0 +1,323 @@
+    - Dapr CLI and Dapr runtime ([instructions](https://docs.dapr.io/getting-started/install-dapr-selfhost/))
+    - Install Azure CLI
+      - Linux ([instructions](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli-linux?pivots=apt))
+      - macOS ([instructions](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli-macos))
+      - Windows ([instructions](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli-windows?tabs=azure-cli))
+    - Install Bicep extension for VS Code ([instructions](https://marketplace.visualstudio.com/items?itemName=ms-azuretools.vscode-bicep))   
+    - If you're running Windows, you'll need to install a **bash shell** to run some of the commands. Install either the [Git Bash](https://git-scm.com/downloads) client or the [Windows Subsystem for Linux 2](https://docs.microsoft.com/en-us/windows/wsl/install-win10).
+
+    Make sure the following minimum software versions are installed by executing the commands in the following table:
+
+    | Software             | Version | Command Line       |
+    | -------------------- | ------- | ------------------ |
+    | Dapr runtime version | v1.2.2  | ```dapr --version```   |
+    | Dapr CLI version     | v1.2.0  | ```dapr --version```   |
+    | DotNet version       | 5.0.302 | ```dotnet --version``` |
+    | azure-cli            | 2.24.0  | ```az --version```     |
+
+### Create Azure Resources
+
+This WhatTheHack will create the following Azure resources in your Azure resource group. Make sure you can create the following:
+
+- Application Insights
+- Azure Kubernetes Service
+- Azure Container Registry
+- Event Hub Namespace
+- IoT Hub
+- KeyVault
+- Log Analytics Workspace
+- Logic App (with the Office 365 activity for sending email)
+- Azure Cache for Redis
+- Storage Account
+- Service Bus Namespace
+
+*If you can't instantiate some of these resources, you won't be able to complete the part of the challenge that uses them, but you may still be able to complete the other challenges*
+
+#### Special Considerations for Azure Kubernetes Service (AKS)
+
+- AKS requires the ability to create a public IP address. This is blocked by many organizations. You will either need to get an exception or have an admin create the AKS cluster for you.
+- The `Resources\Infrastructure\bicep\aks.bicep` file specifies the default values for the cluster that will work for this hack. Customize as needed.
+  - 1 Agent Pool with 3 Linux VMs using the **Standard_DS2_v2** SKU.
+  - 3 services using a total of `300m` of CPU & `300Mi` of memory by default, limited to a total of `3000m` of CPU & `600Mi` of memory.
+  - 1 Zipkin service running to monitor communciation between the services.
+- For simplicity, a Kubernetes secret is used to allow AKS to pull images from the Azure Container Registry. **This is not best practice**. In a production example, you should use a managed identity & RBAC.
```

Comment:
This should be MORE onerous!

I'd put a bold "Warning" label on it like this:

"**WARNING:** For simplicity....

Reply:
Change "This is not best practice" to "This is not a best practice"